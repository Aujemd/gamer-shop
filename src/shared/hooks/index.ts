export { useStoredCart } from "./use-stored-cart";
