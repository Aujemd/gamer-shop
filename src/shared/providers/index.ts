export * from "./data-fetching-provider";
