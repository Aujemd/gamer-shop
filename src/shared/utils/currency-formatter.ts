/**
 * Formats a number as a currency string.
 * @param amount - The amount to format.
 * @param locale - The locale to use for formatting (default is "en-US").
 * @param currency - The currency code to use for formatting (default is "USD").
 * @returns The formatted currency string.
 */
export function formatCurrency(amount: number, locale: string = "en-US", currency: string = "USD"): string {
    return new Intl.NumberFormat(locale, {
        style: "currency",
        currency,
    }).format(amount);
}
