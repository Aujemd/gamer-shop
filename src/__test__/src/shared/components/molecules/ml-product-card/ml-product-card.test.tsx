import { render, screen } from "@testing-library/react";

import type { Game } from "@/utils/endpoint";

import { formatCurrency, MlProductCard } from "@/shared";
import { allGames } from "@/utils/endpoint";

const mockGame: Game = allGames[0];

describe("mlProductCard", () => {
    it("should render correctly", () => {
        const { asFragment } = render(
            <MlProductCard
                data-testid="ml-product-card-test-id"
                {...mockGame}
            />,
        );
        expect(asFragment()).toMatchSnapshot();
    });
    it("renders the card with all main elements", () => {
        render(
            <MlProductCard
                data-testid="ml-product-card-test-id"
                {...mockGame}
            />,
        );
        expect(screen.getByTestId("ml-product-card-test-id")).toBeInTheDocument();
        expect(screen.getByTestId(`ml-product-card-image-test-id-${mockGame.id}`)).toBeInTheDocument();
        expect(screen.getByTestId(`ml-product-card-genre-test-id-${mockGame.id}`)).toHaveTextContent(mockGame.genre);
        expect(screen.getByTestId(`ml-product-card-name-test-id-${mockGame.id}`)).toHaveTextContent(mockGame.name);
        expect(screen.getByTestId(`ml-product-card-price-test-id-${mockGame.id}`)).toHaveTextContent(formatCurrency(mockGame.price));
        expect(screen.getByTestId(`ml-product-card-is-new-label-test-id-${mockGame.id}`)).toBeInTheDocument();
    });

    it("does not render the 'New' badge if isNew is false", () => {
        render(<MlProductCard {...mockGame} isNew={false} />);
        expect(screen.queryByTestId(`ml-product-card-is-new-label-test-id-${mockGame.id}`)).not.toBeInTheDocument();
    });

    it("applies custom className", () => {
        render(
            <MlProductCard
                {...mockGame}
                data-testid="ml-product-card-test-id"
                className="custom-class"
            />,
        );
        expect(screen.getByTestId("ml-product-card-test-id")).toHaveClass("custom-class");
    });
});
