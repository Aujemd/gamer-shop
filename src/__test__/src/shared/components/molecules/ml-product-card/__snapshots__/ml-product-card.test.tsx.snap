// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mlProductCard should render correctly 1`] = `
<DocumentFragment>
  <article
    class="p-6 space-y-5 border-half border-stroke-secondary rounded-2xl font-archivo "
    data-testid="ml-product-card-test-id"
    description="An open-world, action-adventure story set in Night City."
  >
    <div
      class="relative h-60"
    >
      <img
        alt="Cyberpunk 2077"
        class="object-cover rounded-t-2xl"
        data-nimg="fill"
        data-testid="ml-product-card-image-test-id-1"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=%2Fgame-images%2Fcyberpunk2077.jpeg&w=3840&q=75"
        srcset="/_next/image?url=%2Fgame-images%2Fcyberpunk2077.jpeg&w=640&q=75 640w, /_next/image?url=%2Fgame-images%2Fcyberpunk2077.jpeg&w=750&q=75 750w, /_next/image?url=%2Fgame-images%2Fcyberpunk2077.jpeg&w=828&q=75 828w, /_next/image?url=%2Fgame-images%2Fcyberpunk2077.jpeg&w=1080&q=75 1080w, /_next/image?url=%2Fgame-images%2Fcyberpunk2077.jpeg&w=1200&q=75 1200w, /_next/image?url=%2Fgame-images%2Fcyberpunk2077.jpeg&w=1920&q=75 1920w, /_next/image?url=%2Fgame-images%2Fcyberpunk2077.jpeg&w=2048&q=75 2048w, /_next/image?url=%2Fgame-images%2Fcyberpunk2077.jpeg&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
      <span
        aria-label="New Product"
        class="absolute top-3 left-3 bg-stone-100 text-gray-medium px-3 py-2 rounded-sm text-center capitalize tracking-sm leading-4 text-base font-normal"
        data-testid="ml-product-card-is-new-label-test-id-1"
      >
        New
      </span>
    </div>
    <div
      class="space-y-3"
    >
      <span
        class="uppercase font-bold text-base leading-4 text-neutral-500 tracking-normal block"
        data-testid="ml-product-card-genre-test-id-1"
      >
        Action
      </span>
      <div
        class="flex items-center justify-between font-bold tracking-sm text-gray-medium"
      >
        <h2
          class="inline text-lg leading-5 whitespace-nowrap text-ellipsis overflow-hidden"
          data-testid="ml-product-card-name-test-id-1"
        >
          Cyberpunk 2077
        </h2>
        <data
          class="inline text-xl leading-6 text-center"
          data-testid="ml-product-card-price-test-id-1"
          value="59.99"
        >
          $59.99
        </data>
      </div>
    </div>
    <button
      aria-label="add Cyberpunk 2077 to cart"
      class="font-archivo font-bold text-base leading-4 tracking-sm text-center rounded-lg py-5 w-full border border-gray-medium uppercase hover:bg-gray-medium hover:text-white cursor-pointer transition-colors"
      data-testid="ml-product-card-button-1"
    >
      ADD TO CART
    </button>
  </article>
</DocumentFragment>
`;
